### PR TITLE
Move tests of `IDE0060` to `VerifyCS`

### PR DESCRIPTION
**Pull request by DoctorKrolic**
- Disabled `IDE0058` and `IDE0059` in these tests since they introduce too much noise
- Fixed some compilation errors, which doesn't seem to be part of the test case
- 1 test is gone, since it is now covered by another case
- There were 2 tests like `void M(_, _)`, where in one case `M` is a method and in another - a local function. I don't understand what value these tests bring since such code is not even syntactically correct. I didn't remove them, but disabled verifying compilation errors in them to not make too much noise
- There were cases when parameter is passed into different method and the test case was to not report diagnostic there. But since the calee method wasn't doing anything with its parameter, it triggered diagnostic. In such cases I added `var x = <parameter>;` to the calee to not shift focus from the main purpose of that particular test case.
- Сompacted raw string markers
- In 2 or 3 cases there was a duplicate `Trait`, that replicated the one above the test class. Removed these duplicates
- Introduced raw strings where interpolation was involved
- In 2 last tests legacy `WorkItem` attribute constructor was used. Fixed. ~~Refactored `WorkItemAttribute` to hide this constructor overload when editing~~

### CODE REVIEW

File: src/Compilers/Test/Core/Assert/WorkItemAttribute.cs [hunk 1]
```diff
@@ -2,9 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
```
Comment (333fred) [inline src/Compilers/Test/Core/Assert/WorkItemAttribute.cs:5]: When the simple ctor was added, we specifically did not hide the existing one. Please revert the changes in this file.

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs [hunk 2]
```diff
@@ -1153,11 +1153,11 @@ void M3(int {|#4:p5|})
                     """,
                 ExpectedDiagnostics =
                 {
-                    VerifyCS.Diagnostic("IDE0060").WithLocation(0).WithMessage("Remove unused parameter 'p1'"),
-                    VerifyCS.Diagnostic("IDE0060").WithLocation(1).WithMessage("Parameter 'p2' can be removed; its initial value is never used"),
-                    VerifyCS.Diagnostic("IDE0060").WithLocation(2).WithMessage("Remove unused parameter 'p3' if it is not part of a shipped public API"),
-                    VerifyCS.Diagnostic("IDE0060").WithLocation(3).WithMessage("Parameter 'p4' can be removed if it is not part of a shipped public API; its initial value is never used"),
-                    VerifyCS.Diagnostic("IDE0060").WithLocation(4).WithMessage("Parameter 'p5' can be removed; its initial value is never used"),
+                    VerifyCS.Diagnostic("IDE0060").WithSeverity(DiagnosticSeverity.Info).WithLocation(0).WithMessage("Remove unused parameter 'p1'"),
```
Comment (sharwell) [inline src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs:1156]: 💡 Use `WithArguments` instead of `WithMessage`. The message format is included in the diagnostic descriptor and does not need to be duplicated here.

**Review by sharwell — CHANGES_REQUESTED** [review]
The content of some tests was changed unexpectedly during this work.

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs [hunk 3]
```diff
@@ -65,30 +51,36 @@ void M(int [|p|])
         [Fact]
         public async Task Parameter_Unused()
         {
-            await TestDiagnosticsAsync(
-                """
+            await VerifyAnalyzerAsync("""
                 class C
                 {
-                    void M(int [|p|])
+                    void M(int {|IDE0060:p|})
```
Comment (sharwell) [inline src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs:57]: 📝 I mentioned in another commit that literal diagnostic IDs should reference the named constants. Note that this would _not_ apply for IDs appearing with the markup syntax `{|IDE####:...|}`.

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs [hunk 4]
```diff
@@ -216,31 +212,28 @@ void M(int [|p|], bool flag, bool flag2)
...
         [Fact]
         public async Task Parameter_UsedInConstructorInitializerOnly()
         {
-            await TestDiagnosticMissingAsync(
-                """
+            await VerifyAnalyzerAsync("""
                 class B
                 {
-                    protected B(int p) { }
+                    protected B(int p) { var x = p; }
```
[the remark this answers is not in the document]
Comment (DoctorKrolic) [reply src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs:231]: I described the reason for the change in the summary of this PR. In this particular case the old test span is around parameter of constructor in the derived class to verify, that the diagnostic isn't reported there. With the new verifier all file is tested now, revealing that this parameter in the base class produces diagnostic as well. I dicided to "disable" diagnostics in such cases with this trick to represent old test behaviour as close as possible and not add additional diagnostic spans, so reader can concentrate on the primary goal of the test without distractions to other "by-product" diagnostics. I can move all this changes to a separate commit, but I don't think this will fulfill the first condition anyway. Therefore I want to ensure that you understood the reason for the change correctly and you are still sure, that such reason isn't enough for making changes in tests. So, should I revert all such changes to tests and replace them with diagnostic markups?

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs [hunk 5]
```diff
@@ -65,30 +51,36 @@ void M(int [|p|])
...
+                    {{typeAccessibility}} class C
+                    {
+                        {{methodAccessibility}} void M(int p)
+                        {
+                        }
+                    }
+                    """,
+                Options =
+                {
+                    { CodeStyleOptions2.UnusedParameters, UnusedParametersPreference.NonPublicMethods, NotificationOption2.Suggestion }
```
[the remark this answers is not in the document]
Comment (DoctorKrolic) [reply src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs:81]: This is a direct copy from old code. But I trust you!

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (DoctorKrolic) [reply src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs:1156]: Unfortunatelly, this is not true for this diagnostic. Here is code that initializes descriptor for it:
https://github.com/dotnet/roslyn/blob/d84d54f2e5d3d66d81453fbd5fafbeeecd0d6cde/src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs#L81-L87
If we go-to-def of `CreateDescriptorWithId`, we will see this:
https://github.com/dotnet/roslyn/blob/d84d54f2e5d3d66d81453fbd5fafbeeecd0d6cde/src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer_Core.cs#L46-L63
Here severity is initialized as hidden, but it is somewhere overriden to info. So using descriptor won't help much, but will force to make it internal in the analyzer

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs:231]: Normally for this situation, we would just add the diagnostic spans which were previously missed by the test.

[the remark this answers is not in the document]
Comment (DoctorKrolic) [reply src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs]: Ok, removed all this "suppressions". However, there are still several cases where I fixed compiler errors, which are not connected with the test, e.g. in one test there was `;` missing, which is 100% not the main purpose of that test. I think, such fixes are still valuable.

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs (hunk 2 above)
Comment (DoctorKrolic) [reply src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs]: Unfortunatelly, `IDE0060` can have multiple message templates and such effect is done by enforsing message here:
https://github.com/dotnet/roslyn/blob/d84d54f2e5d3d66d81453fbd5fafbeeecd0d6cde/src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs#L159-L183
I would say this is a flaw in current design: here message is constucted with arguments inside, so these arguments are not considered to be arguments of diagnostic. So I tried to use `WithArguments` and it failed.

[the remark this answers is not in the document]
Comment (sharwell) [reply src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs]: It's being overridden by this:

```
{ CodeStyleOptions2.UnusedParameters, UnusedParametersPreference.NonPublicMethods, NotificationOption2.Suggestion }
```

When you switch to the simpler form, it will go back to using the default severity.

Comment (sharwell) [reply src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs]: Weird, but works for me.

**Comment by DoctorKrolic** [comment]
@sharwell PTAL. I've reverted most test changes

**Comment by sharwell** [comment]
@DoctorKrolic Was this just closed as stale? Sorry I missed the earlier update notifications.

**Comment by DoctorKrolic** [comment]
There are 3 reasons why I closed this PR:
1. Yes, it is stale. I don't like keeping such 'zombie' PRs for too long
2. In one of your review comments you pointed out, that changes to tests should be done in a separate commit. Well, achieving it here will require manually finding the rest of test changes and separating them from other changes of the PR. It is quite a bit of work for not so much gain
3. Most importantly, test `RegressionTest_ShouldReportUnusedParameter_02` fail in CI. It is surrounded by `#if` with a comment, that it is 'not applicable for CodeStyle layer as attempting to fetch an editorconfig string representation for this invalid option fails'. I am not sure what it is meant to test then if the option is incorrect, but if we follow the rule "don't remove existing tests" then we are stuck. It isn't possible neither to port it to `VerifyCS` nor to just get rid of it
